addons/mail/static/src/core/common/message_search_hook.js

```diff
@@ -0,0 +1,90 @@
+/* @odoo-module */
+
+import { useSequential } from "@mail/utils/common/hooks";
+import { useState, onWillUnmount, markup } from "@odoo/owl";
+import { useService } from "@web/core/utils/hooks";
+
+/**
+ * @param {string} searchTerm
+ * @param {string} target
+ */
+export function searchHighlight(searchTerm, target) {
+    if (!searchTerm) {
+        return target;
+    }
+    const htmlDoc = new DOMParser().parseFromString(target, "text/html");
+    for (const term of searchTerm.split(" ")) {
+        const matchs = htmlDoc.evaluate(
+            `//*[text()[contains(translate(., '${term.toUpperCase()}', '${term.toLowerCase()}'), '${term.toLowerCase()}')]]`,
+            htmlDoc,
+            null,
+            XPathResult.ORDERED_NODE_SNAPSHOT_TYPE
+        );
+        for (let i = 0; i < matchs.snapshotLength; i++) {
+            const element = matchs.snapshotItem(i);
+            const newNode = [];
+            for (const node of element.childNodes) {
+                if (node.nodeType === Node.TEXT_NODE) {
+                    const span = htmlDoc.createElement("span");
+                    span.innerHTML = node.textContent.replaceAll(
+                        new RegExp(`(${term})`, "gi"),
+                        '<span class="o-mail-Message-searchHighlight">$&</span>'
+                    );
```

Comment:
`term` here should be escaped, as it's mixed with HTML and then injected into `innerHTML`
It's probably not exploitable (self injections don't matter) but it's also necessary for the consistency of the result. You can test with posting `&amp;` and then searching it, it displays `&` but should `&amp;`

There should be a test for that.